vault-kmip-setup.sh

```diff
@@ -0,0 +1,216 @@
+      ;;
+    *)
+      echo "[INFO] Launching new Vault container"
+      docker run -d \
+        --name "${CONTAINER_NAME}" \
+        -e VAULT_DISABLE_MLOCK=true \
+        --cap-add IPC_LOCK \
+        -p 8200:8200 \
+        -p 8201:8201 \
+        -p 5696:5696 \
+        -v "${CERTS_DIR}:/vault/certs" \
+        -v "${VAULT_HCL}:/vault/config/vault.hcl" \
+        -v "${DATA_DIR}:/vault/data" \
+        -v "${LOG_DIR}:/vault/log" \
+        -e "VAULT_LICENSE=$(<"${VAULT_LICENSE}")" \
+        hashicorp/vault-enterprise:latest \
+        vault server -config=/vault/config/vault.hcl >/dev/null
+      ;;
+  esac
+  sleep 10
+  echo "[INFO] Vault container started"
+}
+
+initialize_vault() {
+  local init_file="${CERTS_DIR}/init_token.json"
+
+  if [[ -f "${init_file}" ]]; then
+    echo "[INFO] Vault already initialized at ${init_file}"
+  else
+    echo "[INFO] Initializing Vault..."
+    docker exec "${CONTAINER_NAME}" apk add --no-cache jq >/dev/null
+
+    docker exec "${CONTAINER_NAME}" sh -c '
+      export VAULT_ADDR="http://127.0.0.1:8200"
+      vault operator init -format=json > /vault/certs/init_token.json
+    ' || {
+      echo "[ERROR] Vault initialization failed"
+      return 1
+    }
+    echo "[INFO] Vault initialized."
+  fi
+
+  echo "[INFO] Unsealing Vault..."
+
+  for i in {0..2}; do
+    local unseal_key
+    unseal_key=$(docker exec "${CONTAINER_NAME}" jq -r ".unseal_keys_b64[${i}]" /vault/certs/init_token.json)
+    docker exec "${CONTAINER_NAME}" sh -c "export VAULT_ADDR='http://127.0.0.1:8200' && vault operator unseal '${unseal_key}'" >/dev/null || {
+      echo "[ERROR] Failed to unseal with key ${i}"
+      return 1
+    }
+  done
+
+  echo "[INFO] Vault unsealed successfully"
+}
+
+configure_kmip() {
+  local root_token
+  root_token=$(docker exec "${CONTAINER_NAME}" \
+    jq -rM '.root_token' /vault/certs/init_token.json)
+
+  echo "[INFO] Configuring KMIP secrets engine"
+    docker exec "${CONTAINER_NAME}" sh -c "
+    export VAULT_ADDR='http://127.0.0.1:8200' VAULT_TOKEN='${root_token}'
+    set -e
+
+    if ! vault secrets list | grep -q kmip/; then
+      vault secrets enable kmip
+    fi
+
+    vault write kmip/config \
+      tls_ca_key_type='rsa' \
+      tls_ca_key_bits=2048 \
+      listen_addrs='0.0.0.0:5696' \
+      server_hostnames='172.17.0.1'
+
+    vault read -field=ca_pem kmip/ca > /vault/certs/vault-kmip-ca.pem
+
+    # Check if scope exists before creating
+    if ! vault list kmip/scope 2>/dev/null | grep -q my-service 2>/dev/null; then
+      vault write -f kmip/scope/my-service
+    fi
+
+    vault write kmip/scope/my-service/role/admin \
+      operation_all=true \
+      tls_client_key_bits=2048 \
+      tls_client_key_type=rsa \
+      tls_client_ttl=24h
+
+    vault write -format=json kmip/scope/my-service/role/admin/credential/generate \
+      format=pem > /vault/certs/credential.json
+
+    jq -r '.data.certificate' /vault/certs/credential.json > /vault/certs/mysql-client-cert.pem
```

Comment:
jq dependency must be checked if available on the host where the script will be running or else the script will fail

Reply:
agree just like docker.